### DISCUSSION TITLE
Dynamic Routes with <Link /> from one file

### DISCUSSION THREAD

**Discussion by Manubi** [vercel/next.js#19567] [Help] [answered]
Hi,
I am trying to "centralise" my routes in a routes.ts file. It looks like this:

```
export const Routes = {
    Beta: "/beta",
    PasswordReset: "/passwordreset",
    SignIn: "/signin",
    SignUp: "/signup",
};

export const getUserRoutes = (userId: string) => ({
    Onboarding: `user/${userId}/onboarding`,
    EditSettings: `user/${userId}/editSettings`
});
```

I want to use it like this:

```
                        <Link href={getUserRoutes(authUser.id).EditSettings}>
                                        <a>
                                           settings
                                        </a>
                        </Link>
```

but I get the following error:
![Impfi_-_Alles_rund_ums_Impfen](https://user-images.githubusercontent.com/20935138/100366880-b4b82680-3001-11eb-8818-f7095de6b725.png)

I know I can do it like this:
![next_link___Next_js](https://user-images.githubusercontent.com/20935138/100366918-c4376f80-3001-11eb-8a5b-324a373e15ec.png)

But if a route changes I don't want to change it everywhere.... I am looking for an elegant solution, where I only need to change the dynamic route in one file.

Thanks for your help!

**Comment by Tomburgs** [comment, marked as answer]
It seems to me like you forgot to prefix the user routes with a slash?
i.e `/user/${userId}/onboarding`, not `user/${userId}/onboarding`, otherwise you will navigate to a route relative from your current route.

**Reply by Manubi** [reply]
Thanks! You are right... I shouldn't forget to take breaks and go for a walk :)
